crates/shared/src/lib.rs

```diff
@@ -106,6 +106,7 @@ macro_rules! shared_api {
         }
 
         struct Function<'a> {
+            arg_names: Vec<String>,
```

Comment:
If possible it'd be great to use `&'a str` here to allow the encoder/decoder to be pretty lean and avoid the extra `String` allocations

Reply:
I wish! syn's Ident doesn't expose &str and only implements ToString.